tests/unit/test_execution_environment_utils.py

```diff
@@ -0,0 +1,186 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import logging
+import os
+
+from galaxy_importer.utils import execution_environment as ee_utils
+
+
+logger = logging.getLogger(__name__)
+EE_YAML = """
+version: 1
+dependencies:
+  python: requirements.txt
+  system: bindep.txt
+"""
+REQUIREMENTS_TXT = """
+google-auth>=1.0.1  # Apache-2.0
+requests # Apache-2.0
+requests-oauthlib # ISC
+"""
+BINDEP_TXT = """
+# This is a cross-platform list tracking distribution packages needed by tests;
+# see https://docs.openstack.org/infra/bindep/ for additional information.
+
+libffi-devel [platform:rpm]
+python3-devel [compile test platform:rpm]
+"""
+
+
+def test_process_execution_environment():
+    dir = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+    os.mkdir(os.path.join(dir, 'meta'))
+    files = [
+        os.path.join(dir, 'meta/execution-environment.yml'),
+        os.path.join(dir, 'requirements.txt'),
+        os.path.join(dir, 'bindep.txt')
+    ]
+    file_contents = [EE_YAML, REQUIREMENTS_TXT, BINDEP_TXT]
+    try:
+        for f, c in zip(files, file_contents):
+            with open(f, 'w') as fp:
+                fp.write(c)
+                fp.flush()
+        res = ee_utils.process_execution_environment(dir, logger)
+        assert res == {
+            'dependencies': {
+                'python': [
+                    'google-auth>=1.0.1  # Apache-2.0',
+                    'requests # Apache-2.0',
+                    'requests-oauthlib # ISC',
+                ],
+                'system': [
+                    'libffi-devel [platform:rpm]',
+                    'python3-devel [compile test platform:rpm]'
+                ]
+            }
+        }
+    finally:
+        for f in files:
+            os.remove(f)
+        os.rmdir(os.path.join(dir, 'meta'))
+
+
+def test_process_execution_environment_no_python():
```

Comment:
what is the resulting dict supposed to look like if `meta/execution-environment.yml` does not say there are python dependencies, but the python dependencies file `requirements.txt` exists (and vice versa for system deps)... can we have tests that cover this?